cql3/column_condition.cc

```diff
@@ -72,15 +72,12 @@ int is_satisfied_by(cql3::expr::oper_t op, const abstract_type& cell_type,
         }
 }
 
-// Read the list index from key and check that list index is not
-// negative. The negative range check repeats Cassandra behaviour.
-uint32_t read_and_check_list_index(const cql3::raw_value_view& key) {
+// Read the list index from key. Negative values are allowed, like
```

Comment:
I know this has already been decided, but I want to point out again that all these error cases like `list[-1]` being an error, were useful for catching buggy applications. Now `list[-1]` evaluates to null it will be harder for people who write applications to find bugs in the application.

Reply:
It's part of the overall SQL philosophy. I propose we embrace it fully rather than find reasons to be different.